botnim/kb/base.py

```diff
@@ -0,0 +1,31 @@
+from abc import ABC, abstractmethod
+from typing import List, Union, BinaryIO, Tuple
+from ..config import get_logger
+
+logger = get_logger(__name__)
+
+class VectorStore(ABC):
+    """Abstract base class for vector store implementations"""
+    
+    def __init__(self, production: bool = False):
+        self.production = production
+
+    @abstractmethod
+    def create(self, name: str) -> str:
+        """Create a new vector store and return its ID"""
+        pass
+
+    @abstractmethod
+    def upload_documents(self, kb_id: str, documents: List[Union[BinaryIO, Tuple[str, BinaryIO, str]]]) -> None:
+        """Upload documents to the vector store"""
+        pass
+
+    @abstractmethod
```

Comment:
is this being used anywhere?

Reply:
not yet, but I want to keep it a placeholder for future usage.